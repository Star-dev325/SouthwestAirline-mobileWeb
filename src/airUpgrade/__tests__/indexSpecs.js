import React from 'react';
import { AirUpgrade } from '../index';
import { shallow } from 'enzyme';

describe('airUpgrade Index', () => {
  it('should render default props', () => {
    const airUpgradeWrapper = createComponent();

    expect(airUpgradeWrapper).toMatchSnapshot();
  });

  it('should render airUpgrade component routes when AIR_UPGRADE is true', () => {
    const airUpgradeWrapper = createComponent({ AIR_UPGRADE: true });

    expect(airUpgradeWrapper).toMatchSnapshot();
  });

  const createComponent = (props) => {
    const defaultProps = {
      match: {
        url: 'air/upgrade'
      },
      AIR_UPGRADE: false
    };
    const combinedProps = {
      ...defaultProps,
      ...props
    };

    return shallow(<AirUpgrade {...combinedProps} />);
  };
});
