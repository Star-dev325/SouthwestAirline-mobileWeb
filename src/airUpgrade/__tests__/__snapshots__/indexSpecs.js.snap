// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`airUpgrade Index should render airUpgrade component routes when AIR_UPGRADE is true 1`] = `
<div>
  <Route
    component={[Function]}
    exact={true}
    path="air/upgrade"
  />
  <Route
    component={[Function]}
    exact={true}
    path="air/upgrade/select-bounds"
  />
  <Route
    component={[Function]}
    exact={true}
    path="air/upgrade/index.html"
  />
  <Route
    component={[Function]}
    exact={true}
    path="air/upgrade/select.html"
  />
  <Route
    component={[Function]}
    exact={true}
    path="air/upgrade/purchase/paypal-canceled"
  />
  <Route
    component={[Function]}
    exact={true}
    path="air/upgrade/purchase/paypal"
  />
  <Route
    component={[Function]}
    exact={true}
    path="air/upgrade/purchase.html"
  />
  <Route
    component={[Function]}
    exact={true}
    path="air/upgrade/confirmation.html"
  />
  <Route
    component={[Function]}
    exact={true}
    path="air/upgrade/payment"
  />
  <Route
    component={[Function]}
    exact={true}
    path="air/upgrade/contact-method"
  />
  <Route
    component={[Function]}
    exact={true}
    path="air/upgrade/apply-travel-funds"
  />
</div>
`;

exports[`airUpgrade Index should render default props 1`] = `
<div>
  <Redirect
    to="/"
  />
</div>
`;
